core/cloudflow-operator/src/main/scala/cloudflow/operator/action/Action.scala

```diff
@@ -160,31 +159,30 @@ class CreateAction[T <: ObjectResource](
 /**
  * Captures the update of the resource.
  */
-class UpdateAction[T <: ObjectResource](
+class CreateOrPatchAction[T <: ObjectResource, O <: Patch](
     val resource: T,
+    val patch: O,
     implicit val format: Format[T],
-    implicit val resourceDefinition: ResourceDefinition[T],
-    val editor: ObjectEditor[T]
+    implicit val patchWriter: Writes[O],
+    implicit val resourceDefinition: ResourceDefinition[T]
 ) extends Action[T] {
 
-  val name = "update"
+  val name = "create-or-patch"
 
   /**
    * Updates the resource, without changing the `resourceVersion`.
    */
   def execute(client: KubernetesClient)(implicit ec: ExecutionContext, lc: LoggingContext): Future[Action[T]] =
     for {
-      result <- executeUpdate(client)
-    } yield new UpdateAction(result, format, resourceDefinition, editor)
-  private def executeUpdate(client: KubernetesClient)(implicit ec: ExecutionContext, lc: LoggingContext): Future[T] =
+      result <- executeCreateOrPatch(client)
+    } yield new CreateOrPatchAction(result, patch, format, patchWriter, resourceDefinition)
+
+  private def executeCreateOrPatch(client: KubernetesClient)(implicit ec: ExecutionContext, lc: LoggingContext): Future[T] =
     for {
       existing ← client.getOption[T](resource.name)
       res ← existing
-        .map(existingResource ⇒
-          editor.updateMetadata(resource, resource.metadata.copy(resourceVersion = existingResource.metadata.resourceVersion))
-        )
-        .map(resourceToUpdate ⇒ recoverFromConflict(client.update(resourceToUpdate), client, executeUpdate))
-        .getOrElse(client.create(resource))
+        .map(_ ⇒ client.patch(resource.name, patch, Some(resource.ns)))
+        .getOrElse(recoverFromConflict(client.create(resource), client, executeCreateOrPatch))
```

Comment:
this might be a tricky situation. the `patch` could also cause a conflict, no?

Reply:
good catch! yes it gets a little tricky

Reply:
fixed.